src/mono/mono/utils/atomic.h

```diff
@@ -457,6 +512,8 @@ static inline void mono_atomic_store_i64(volatile gint64 *dst, gint64 val)
 
 #define WAPI_NO_ATOMIC_ASM
 
+extern guint8 mono_atomic_cas_u8(volatile guint8 *dest, guint8 exch, guint8 comp);
```

Comment:
It's interesting that we don't need implementations for these in atomic.c. I guess we don't have any platforms with emulated atomics anymore